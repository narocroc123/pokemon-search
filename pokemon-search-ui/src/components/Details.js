import styled from "styled-components";
import { useSelector } from "react-redux";

const LegendaryTitle = styled.h1`
  font-style: italic;
`;

const LegendaryLabel = styled.h3`
  color: #0066a2;
`;

const Details = () => {
  const searchResult = useSelector((state) => state.search);

  const name = searchResult.data.name.toUpperCase();
  const description = searchResult.data.description;

  return (
    <>
      {searchResult.data.is_legendary ? (
        <>
          <LegendaryTitle>{name}</LegendaryTitle>
          <LegendaryLabel>Legendary!</LegendaryLabel>
        </>
      ) : (
        <h1>{name}</h1>
      )}
      <p>{description}</p>
    </>
  );
};

export default Details;
