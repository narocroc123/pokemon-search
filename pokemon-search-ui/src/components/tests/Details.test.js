import React from "react";
import { render } from "@testing-library/react";

import Details from "../Details";
import { Provider } from "react-redux";
import { createStore } from "redux";
import rootReducer from "../../reducers/index";

const preLoadedState = {
  search: {
    data: {
      name: "charizard",
      description:
        "Spits fire that is hot enough to melt boulders. Known to cause forest fires unintentionally.",
      is_legendary: true,
    },
  },
  history: ["pikachu", "azelf", "charizard"],
};

const mockStore = createStore(rootReducer, preLoadedState);

describe("App", () => {
  const component = (
    <Provider store={mockStore}>
      <Details />
    </Provider>
  );

  test("renders App component", () => {
    render(component);
  });

  test("render to match snapshot", () => {
    const { asFragment } = render(component);

    expect(asFragment(component)).toMatchSnapshot();
  });

  test("contains pokemon name", () => {
    const { getByText } = render(component);

    expect(getByText("CHARIZARD")).toBeInTheDocument();
  });

  test("contains pokemon description", () => {
    const { getByText } = render(component);

    expect(
      getByText(
        "Spits fire that is hot enough to melt boulders. Known to cause forest fires unintentionally."
      )
    ).toBeInTheDocument();
  });

  test("contains pokemon legendary label", () => {
    const { getByText } = render(component);

    expect(getByText("Legendary!")).toBeInTheDocument();
  });
});
