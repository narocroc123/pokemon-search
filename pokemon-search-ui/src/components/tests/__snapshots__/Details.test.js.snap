// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render to match snapshot 1`] = `
<DocumentFragment>
  <h1
    class="sc-bczRLJ bTRrLY"
  >
    CHARIZARD
  </h1>
  <h3
    class="sc-gsnTZi iTPgsc"
  >
    Legendary!
  </h3>
  <p>
    Spits fire that is hot enough to melt boulders. Known to cause forest fires unintentionally.
  </p>
</DocumentFragment>
`;
